Update metrics names

sradco opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as a draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
#### Before this PR:
Recording rule kubevirt_vmi_vcpu_count didn't meet with the naming best practices.

#### After this PR:
This PR updated the name of kubevirt_vmi_vcpu_count to vmi:kubevirt_vmi_vcpu:count,
so that the comply to the naming best practices.

### References
<!-- optional,
  Use `Fixes #<issue number>(, Fixes #<issue_number>, ...)` format, to close the issue(s) when PR gets merged.
  Use `Partially addresses #<issue number>` to link an issue without closing it when the PR merges.
  
- Partially addresses #
-->
- Fixes #https://issues.redhat.com/browse/CNV-71409

<!-- optional,
  VEP tracking issue if this PR is implementing one.
  For additional info about VEP tracking issue, see https://github.com/kubevirt/enhancements#process
  
- VEP tracking issue: https://github.com/kubevirt/enhancements/issues/<vep_tracking_issue_number>
-->

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least one e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Recording rule kubevirt_vmi_vcpu_count name changes to vmi:kubevirt_vmi_vcpu:count
```

avlitman commented on tests/libmigration/migration.go:
```diff
@@ -178,7 +178,6 @@ func ConfirmMigrationDataIsStored(virtClient kubecli.KubevirtClient, migration *
 	Expect(vmi.Status.MigrationState.Completed).To(Equal(migration.Status.MigrationState.Completed), "the VMI and migration completed state should be equal")
 	Expect(vmi.Status.MigrationState.Failed).To(Equal(migration.Status.MigrationState.Failed), "the VMI nad migration failed status must be equal")
 	Expect(vmi.Status.MigrationState.MigrationUID).To(Equal(migration.Status.MigrationState.MigrationUID), "the VMI migration UID and the migration object UID should match")
-	return
```
nit: it's not needed but maybe better to remove not it this pr not sure

sradco committed:
Update kubevirt_vmi_vcpu_count name

Update kubevirt_vmi_vcpu_count name
to vmi:kubevirt_vmi_vcpu:count to comply
with recording rules naming conventions.

Also, Add cluster:kubevirt_virt_controller_pods_running:coun
to the exclusMetrics list so that it will not fail
the metrics linter.

Signed-off-by: Shirly Radco <sradco@redhat.com>

enp0s3 commented:
/override pull-kubevirt-metrics-lint

sradco commented:
@enp0s3 I had to add kubevirt_vmi_migration_data_total_bytes to the excludeMetrics so that the linter will not fail.
Please review again

avlitman commented:
/lgtm

sradco commented:
/test pull-kubevirt-fuzz

machadovilaca commented:
@sradco https://github.com/kubevirt/kubevirt/pull/15988 is merged
but maybe you need to rebase

/retest

sradco commented:
/unhold
